server/service/global_policies.go

```diff
@@ -158,19 +167,41 @@ func deleteGlobalPoliciesEndpoint(ctx context.Context, request interface{}, svc
 // DeleteGlobalPolicies deletes the given policies from the database.
 // It also deletes the given ids from the failing policies webhook configuration.
 func (svc Service) DeleteGlobalPolicies(ctx context.Context, ids []uint) ([]uint, error) {
-	if err := svc.authz.Authorize(ctx, &fleet.Policy{}, fleet.ActionWrite); err != nil {
+	// First check if authorized to read policies
+	if err := svc.authz.Authorize(ctx, &fleet.Policy{}, fleet.ActionRead); err != nil {
 		return nil, err
 	}
 	if len(ids) == 0 {
 		return nil, nil
 	}
+	policiesByID, err := svc.ds.PoliciesByID(ctx, ids)
```

Comment:
@lucasmrod, I like the idea of keeping things simple, although in this case I wonder if we might be better off in the long run having this be datastore method that retrieves a map of full policy rather than just names. My thinking was that we could avoid expanding into a variety of "...FromIDs" methods for each field of the policy struct.

I wouldn't usually try to do more than is needed for the current ticket ,but here it seemed pretty likely that we would need a general purpose method like this again for future tasks. I'm happy to go with whatever so just let me know :)

Reply:
1. Agree, `PolicyByIDs` is a valid method as is and will be useful for devs wanting to load a set of polices (full load) from their ids.
2. It's an "expensive" method, but given that it's used on the API/UI side (not on the hosts side) this is ok. If we were to call this new `Datastore` method on the hosts/osquery-agents side (`server/service/service_osquery.go`) then the scale is different and the sub-selects to `policy_membership` would generate a lot of load. 

So, all things considered, am ok with keeping it as is.

PS re 2: We are thinking on solving this "potential issue" by decoupling the current `Datastore` into two separate data stores, one for API/UI usage and another "lite" datastore for osquery agents (we've learned some of this during the 4.9.0 refactor).